chromium/rules.js

```diff
@@ -405,51 +416,61 @@ RuleSets.prototype = {
    */
   potentiallyApplicableRulesets: function(host) {
     // Have we cached this result? If so, return it!
-    var cached_item = this.ruleCache.get(host);
-    if (cached_item !== undefined) {
-      log(DBUG, "Ruleset cache hit for " + host + " items:" + cached_item.length);
+    if (this.ruleCache.has(host)) {
+      let cached_item = this.ruleCache.get(host);
+      let len = cached_item && cached_item != nullIterable ? cached_item.size : 0;
+      log(DBUG, "Ruleset cache hit for " + host + " items:" + len);
       return cached_item;
+    } else {
+      log(DBUG, "Ruleset cache miss for " + host);
     }
-    log(DBUG, "Ruleset cache miss for " + host);
 
-    var results = [];
+    // Let's search!
+    let resultSet = new Set();
+
     if (this.targets.has(host)) {
       // Copy the host targets so we don't modify them.
-      results = results.concat(this.targets.get(host));
+      let results = this.targets.get(host);
+      results.map(result => resultSet.add(result));
     }
 
     // Ensure host is well-formed (RFC 1035)
-    if (host.indexOf("..") != -1 || host.length > 255) {
-      log(WARN,"Malformed host passed to potentiallyApplicableRulesets: " + host);
-      return null;
+    if (host.length > 255 || host.indexOf("..") != -1) {
+      log(WARN, "Malformed host passed to potentiallyApplicableRulesets: " + host);
+      return nullIterable;
     }
 
     // Replace each portion of the domain with a * in turn
-    var segmented = host.split(".");
-    for (let i=0; i < segmented.length; i++) {
-      let tmp = segmented[i];
-      segmented[i] = "*";
-      results = results.concat(this.targets.get(segmented.join(".")));
```

Comment:
This could be something like my above comment, (https://github.com/EFForg/https-everywhere/pull/12577/files#r139258106) or maybe a ternary expression like
```javascript
results = (this.targets.has(segmented.join('.')) ? 
  (new Set([...results, ...this.targets.get(segmented.join('.')]) :
  results);
```